homeassistant/components/airtouch4/climate.py

```diff
@@ -0,0 +1,338 @@
+
+    async_add_devices(all_devices)
+
+
+class AirtouchAC(ClimateEntity, CoordinatorEntity):
+    """Representation of an AirTouch 4 ac."""
+
+    def __init__(self, coordinator, ac_number, info):
+        """Initialize the climate device."""
+        super().__init__(coordinator)
+        self._ac_number = ac_number
+        self._airtouch = coordinator.airtouch
+        self._info = info
+        self._unit = self._airtouch.GetAcs()[self._ac_number]
+
+    @callback
+    def _handle_coordinator_update(self):
+        self._unit = self._airtouch.GetAcs()[self._ac_number]
+        return super()._handle_coordinator_update()
+
+    @property
+    def device_info(self):
+        """Return device info for this device."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Airtouch",
+            "model": "Airtouch 4",
+        }
+
+    @property
+    def unique_id(self):
+        """Return unique ID for this device."""
+        return "AC" + str(self._ac_number)
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FAN_MODE
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._unit.Temperature
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return "AC " + str(self._ac_number)
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def fan_mode(self):
+        """Return fan mode of the AC this group belongs to."""
+        return AT_TO_HA_FAN_SPEED[self._airtouch.acs[self._ac_number].AcFanSpeed]
+
+    @property
+    def fan_modes(self):
+        """Return the list of available fan modes."""
+        airtouch_fan_speeds = self._airtouch.GetSupportedFanSpeedsForAc(self._ac_number)
+        return [AT_TO_HA_FAN_SPEED[speed] for speed in airtouch_fan_speeds]
+
+    @property
+    def hvac_mode(self):
+        """Return hvac target hvac state."""
+        is_off = self._unit.PowerState == "Off"
+        if is_off:
+            return HVAC_MODE_OFF
+
+        return AT_TO_HA_STATE[self._airtouch.acs[self._ac_number].AcMode]
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available operation modes."""
+        airtouch_modes = self._airtouch.GetSupportedCoolingModesForAc(self._ac_number)
+        modes = [AT_TO_HA_STATE[mode] for mode in airtouch_modes]
+        modes.extend([HVAC_MODE_OFF])
+        return modes
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new operation mode."""
+        if hvac_mode not in HA_STATE_TO_AT.keys():
+            raise ValueError("HVAC mode not supported")
+
+        if hvac_mode == HVAC_MODE_OFF:
+            return await self.async_turn_off()
+        if self.hvac_mode == HVAC_MODE_OFF:
```

Comment:
We normally avoid guarding a service with the current state for polling integrations since that might stop the user from actuating the device if we would have an incorrect state.